docs/presentation/View-Models.md

```diff
@@ -0,0 +1,129 @@
+View Models contains the state of your view.
+
+```dart
+@injectable
+class SplashViewModel extends ViewModel implements Initializable, FirstRenderable {
+  @override
+  Future<void> onInitialize() async {
+    // your initialization logic after the view model was created
+  }
+
+  @override
+  Future<void> onFirstRender() async {
+    // your initialization logic after the first frame has rendered
+  }
+}
+```
+
+- The `ViewModel` class inherits from `ChangeNotifier`.
+- A new instance is returned always whenever we resolve a view model.
+- To know how to extract parameters passed during navigation, read about it [here](navigation).
+
+<div style="">
+
+:bulb: **<span style="color: green">TIP</span>**
```

Comment:
I like your use of `TIP`, `AVOID`, etc patterns. As it provides a quick entry into the most relevant parts of the docs.